client/boot/index.js

```diff
@@ -228,6 +231,32 @@ function reduxStoreReady( reduxStore ) {
 	// Layout themselves when logged in.
 	if ( ! isIsomorphic || user.get() ) {
 		renderLayout( reduxStore );
+
+		if ( config.isEnabled( 'catch-js-errors' ) ) {
+			const Logger = require( 'lib/catch-js-errors' );
+			const errorLogger = new Logger();
+			//Save data to JS error logger
+			errorLogger.saveDiagnosticData( {
+				user_id: user.get().ID,
+				calypso_env: config( 'env_id' )
+			} );
+			errorLogger.saveDiagnosticReducer( function() {
+				const state = reduxStore.getState();
+				return {
+					blog_id: getSelectedSiteId( state ),
+					calypso_section: getSectionName( state )
+				};
+			} );
+			errorLogger.saveDiagnosticReducer( () => ( { tests: getSavedVariations() } ) );
+			analytics.on( 'record-event', ( eventName, eventProperties ) => errorLogger.saveExtraData( { lastTracksEvent: eventProperties } ) );
+			page( '*', function( context, next ) {
+				errorLogger.saveNewPath( context.canonicalPath.replace(
+					new RegExp( route.getSiteFragment( context.canonicalPath ), 'g' ),
```

Comment:
Yeah, you're right.
There are some routes that have domain multiple times in them - especially in domain management, but its better to be on the safe side.